### ISSUE TITLE
The same pipeline,and same train data , the train accuracy of each training deviation is very huge

### ISSUE THREAD

**Issue by franklydog** [JohnSnowLabs/spark-nlp#2748]
At present, I have encountered a very strange problem, using the same training data, but the accuracy deviation of each classifier is very much. 
The pipeline is DocumentAssembler-->UniversalSentenceEncoder-->ClassifierDLApproach. 
            val classifierDL = new ClassifierDLApproach()
                .setInputCols("sentence_embeddings")
                .setOutputCol("class")
                .setLabelColumn("category")
                .setEnableOutputLogs(true)
                .setBatchSize(64)
                .setMaxEpochs(20)
                .setLr(5e-3f)
                .setDropout(0.5f)

I get accuracy from the classifierDL log .it's like this:
		Training started - epochs: 20 - learning_rate: 0.005 - batch_size: 64 - training_examples: 35153 - classes: 9
		Epoch 1/20 - 5.24s - loss: 963.88617 - acc: 0.61246717 - batches: 550
		Epoch 2/20 - 4.67s - loss: 878.48175 - acc: 0.7660569 - batches: 550
		Epoch 3/20 - 4.64s - loss: 844.018 - acc: 0.83040684 - batches: 550
		Epoch 4/20 - 4.67s - loss: 842.3266 - acc: 0.83114684 - batches: 550
		Epoch 5/20 - 4.66s - loss: 841.8461 - acc: 0.8316022 - batches: 550
		Epoch 6/20 - 4.65s - loss: 841.7339 - acc: 0.8319153 - batches: 550
		Epoch 7/20 - 4.67s - loss: 841.62585 - acc: 0.83217144 - batches: 550
		Epoch 8/20 - 4.65s - loss: 841.53815 - acc: 0.83242756 - batches: 550
		Epoch 9/20 - 4.67s - loss: 841.0117 - acc: 0.8326268 - batches: 550
		Epoch 10/20 - 4.66s - loss: 840.45447 - acc: 0.83268374 - batches: 550
		Epoch 11/20 - 4.66s - loss: 840.37366 - acc: 0.83274066 - batches: 550
		Epoch 12/20 - 4.67s - loss: 840.3208 - acc: 0.8328545 - batches: 550
		Epoch 13/20 - 4.68s - loss: 840.29205 - acc: 0.8330537 - batches: 550
		Epoch 14/20 - 4.64s - loss: 840.2445 - acc: 0.83311063 - batches: 550
		Epoch 15/20 - 4.67s - loss: 840.1984 - acc: 0.83316755 - batches: 550
		Epoch 16/20 - 4.66s - loss: 840.14703 - acc: 0.83319604 - batches: 550
		Epoch 17/20 - 4.64s - loss: 840.0988 - acc: 0.8332245 - batches: 550
		Epoch 18/20 - 4.66s - loss: 840.0647 - acc: 0.8333099 - batches: 550
		Epoch 19/20 - 4.64s - loss: 840.04706 - acc: 0.83342373 - batches: 550
		Epoch 20/20 - 4.67s - loss: 840.00616 - acc: 0.83345217 - batches: 550

Using the same train sample, I trained nine times, every time consumed about 20 minutes. the all training accuracy is as follows:
a: 0.29696262
b:0.9620615
c: 0.98267347
d: 0.89708525
e: 0.89442694
f: 0.8945768
g:0.896608
h:0.29702717

What is the reason for this? According to the same training data, the trained model should have the same training accuracy.But the results are quite different.

**Comment by maziyarpanahi** [comment]
1- The metrics during the training are the overall quality of the model, not just a few classes, all of them together. Like any other ML/DL some classes might do much better or much worse than the others but the average of all is the metrics during the training.

2- This is not an issue of one particular classifier, this happens with any multi-class multi-label text classification.
- you don't have enough training examples for those classes
- those classes are very hard to predict
- classes are close in terms of meaning (city, location, country, etc.)

This is something that has to be solved in pre-processing, balancing, data augmentation, etc. which all happen before the training and keep adjusting until it becomes even or be tolerated to be accepted.

> Unbalanced Data
Our dataset is unbalanced (it has more samples for some classes than others). This can make the classifier biased toward the one or two classes with lost of samples, while dwarfing others that have less (i.e. the classifier learns the classes with more samples better and remains weak on the smaller classes).

https://medium.com/@b.terryjack/tips-and-tricks-for-multi-class-classification-c184ae1c8ffc
